src/DocSidebar/index.js

```diff
@@ -114,24 +118,26 @@ function DocSidebarItem({theme = 'primary', item, onItemClick, collapsible, ...p
       ItemTag = noLink ? 'span' : Link;
 
       return (
-        <li 
+        <li
           className={classnames(
-            "menu__list-item", 
+            "menu__list-item",
             styles.listItem,
             styles[itemTheme],
             ...getClasses(classNames),
-          )} 
+          )}
           key={label}
         >
           <WithBackgroundImage
             className={classnames("menu__link", styles.menuLink, ...getClasses(iconClasses), {
               [styles.withBackgroundImage]: icon,
             })}
             imageDark={iconDark}
+            imageDarkHover={iconDarkHover}
             imageLight={icon}
+            imageLightHover={iconHover}
             tag={ItemTag}
             to={href}
-            {...(isInternalUrl(href)
+            {...(isInternalUrl(href) && ItemTag !== 'span'
```

Comment:
This feels like we are trying to fit in too much logic. We don't need to fix this now but I need a TODO/FIXME